courseware/api.py

```diff
@@ -46,3 +60,147 @@ def create_edx_user(user):
             user=user, platform=PLATFORM_EDX, has_been_synced=True
         )
     return resp
+
+
+@transaction.atomic
+def create_edx_auth_token(user):
+    """
+    Creates refresh token for LMS for the user
+
+    Args:
+        user(User): the user to create the record for
+
+    Returns:
+        OpenEdXAuth: auth model with refresh_token populated
+    """
+
+    # In order to acquire auth tokens from Open edX we need to perform the following steps:
+    #
+    # 1. Create a persistent session so that state is retained like a browser
+    # 2. Initialize a session cookie for xPro, this emulates a user login
+    # 3. Initiate an Open edX login, delegates to xPro using the session cookie
+    # 4. Initiate an Open edX OAuth2 authorization for xPro
+    # 5. Redirects back to xPro with the access token
+    # 6. Redeem access token for a refresh/access token pair
+
+    # ensure only we can update this for the duration of the
+    auth, _ = OpenEdxApiAuth.objects.select_for_update().get_or_create(user=user)
+
+    # we locked on the previous operation and something else populated these values
+    if auth.refresh_token and auth.access_token:
+        return auth
+
+    # Step 1
+    with requests.Session() as req_session:
+        # Step 2
+        django_session = auth_api.create_user_session(user)
+
+        session_cookie = requests.cookies.create_cookie(
+            name=settings.SESSION_COOKIE_NAME,
+            domain=urlparse(settings.SITE_BASE_URL).hostname,
+            path=settings.SESSION_COOKIE_PATH,
+            value=django_session.session_key,
+        )
+
+        req_session.cookies.set_cookie(session_cookie)
+
+        # Step 3
+        url = edx_url(OPENEDX_SOCIAL_LOGIN_XPRO_PATH)
+        resp = req_session.get(url)
+        resp.raise_for_status()
+
+        # Step 4
+        redirect_uri = urljoin(
+            settings.SITE_BASE_URL, reverse("openedx-private-oauth-complete")
```

Comment:
Yeah, that edx bug has been there since the code was written, probably should issue a PR, but generally if you add `application=None` to that call it fixes it.

Updated the docs to fix the redirect_uri that gets added in edX.